dataclasses_avroschema/fields.py

```diff
@@ -638,6 +645,69 @@ def fake(self) -> typing.Any:
         return self.type.fake()
 
 
+@dataclasses.dataclass
+class DecimalField(BaseField):
+
+    precision: int = -1
+    scale: int = 0
+    has_set_prec_scale = False
+
+    def set_precision_scale(self) -> None:
+        if self.has_set_prec_scale:
+            return
+        self.has_set_prec_scale = True
+        if self.default != types.MissingSentinel:
+            if isinstance(self.default, decimal.Decimal):
+                sign, digits, scale = self.default.as_tuple()
+                self.scale = scale * -1  # Make scale positive, as that's what Avro expects
+                # decimal.Context has a precision property
+                # BUT the precision property is independent of the number of digits stored in the Decimal instance
+                # # # FROM THE DOCS HERE https://docs.python.org/3/library/decimal.html
+                #  The context precision does not affect how many digits are stored.
+                #  That is determined exclusively by the number of digits in value.
+                #  For example, Decimal('3.00000') records all five zeros even if the context precision is only three.
+                # # #
+                # Avro is concerned with *what form the number takes* and not with *handling errors in the Python env*
+                # so we take the number of digits stored in the decimal as Avro precision
+                self.precision = len(digits)
+            elif isinstance(self.default, types.Decimal):
+                self.scale = self.default.scale
+                self.precision = self.default.precision
+            else:
+                raise ValueError("decimal.Decimal default types must be either decimal.Decimal or types.Decimal")
+        else:
+            raise ValueError(
+                "decimal.Decimal default types must be specified to provide precision and scale,"
+                " and must be either decimal.Decimal or types.Decimal"
+            )
+
+            # Just pull the precision from default context and default out scale
+            # Not ideal
+            #
+            # self.precision = decimal.Context().prec
+
+    def get_avro_type(self) -> typing.Dict[str, typing.Any]:
+        self.set_precision_scale()
```

Comment:
this can be done in a `def __post_init__(self)` method, So you will avoid asking whether the values where already parsed (line 659)